### PR TITLE
Eobs 595 refused status visible

### PR DESCRIPTION
**Pull request by Gimpneek**
- prepending 'Refused' to desktop and mobile views

### CODE REVIEW

File: nh_eobs_mental_health/tests/nh_clinical_wardboard/test_read_refused_acuity_index.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+        cls.patch_wardboard_read_super = patch_wardboard_read_super
+
+        def patch_spell_search(*args, **kwargs):
+            return [1]
+
+        def patch_spell_read(*args, **kwargs):
+            context = kwargs.get('context', {})
+            test = context.get('test', '')
+            res = {
```
Comment (Gimpneek) [inline nh_eobs_mental_health/tests/nh_clinical_wardboard/test_read_refused_acuity_index.py]: This is pointless

File: nh_observations/tests/nh_clinical_patient_observation/test_is_patient_refusal_in_effect.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        self.assertEqual(
+            self.observation_model
+                .with_context(test='no_open_obs')
+                .is_patient_refusal_in_effect(1),
+            'full_observation_since_refused_called'
+        )
+
+    def test_open_obs_child(self):
+        """
+        Test returns True if currently open obs is direct child of last refused
```
Comment (Gimpneek) [inline nh_observations/tests/nh_clinical_patient_observation/test_is_patient_refusal_in_effect.py]: Doc string is wrong

**Comment by moderatemisbehaviour** [comment]
:shipit: pending removal of comments and tech debt for performance on patient list.

File: nh_observations/tests/nh_clinical_patient_observation/test_is_patient_refusal_in_effect.py (hunk 2 above)
Comment (Gimpneek) [inline nh_observations/tests/nh_clinical_patient_observation/test_is_patient_refusal_in_effect.py]: Updated

File: nh_eobs_mental_health/tests/nh_clinical_wardboard/test_read_refused_acuity_index.py (hunk 1 above)
Comment (Gimpneek) [inline nh_eobs_mental_health/tests/nh_clinical_wardboard/test_read_refused_acuity_index.py]: Removed

**Comment by moderatemisbehaviour** [comment]
:shipit:

**Comment by moderatemisbehaviour** [comment]
:shipit:

**Comment by Nayira** [comment]
:shipit:  for code review and creation of test machine (if everything is working correctly after merging develop back in)

**Comment by Nayira** [comment]
Review of changes :shipit:

File: nh_eobs/sql_statements.py [hunk 3]
```diff
@@ -352,16 +352,18 @@ def get_workload(self, settings):
                 else false
             end as notification
         from nh_activity activity
-        inner join nh_activity spell on spell.id = activity.parent_id
+        inner join nh_activity spell_activity
```
Comment (Gimpneek) [inline nh_eobs/sql_statements.py]: Renaming this confusing join name as it's the spell activity and not the spell that this represents

File: nh_eobs_mental_health/models/nh_clinical_sql.py [hunk 4]
```diff
@@ -367,29 +367,44 @@ def get_refused_wardboard(self, interval):
 
     def get_collect_activities_sql(self, activity_ids_sql):
         sql = self.collect_activities_skeleton.replace(
-            'left join ews1 on ews1.spell_activity_id = spell.id',
-            'left join ews1 on ews1.spell_activity_id = spell.id '
+            'left join ews1 on ews1.spell_activity_id = spell_activity.id',
+            'left join ews1 on ews1.spell_activity_id = spell_activity.id '
+            'LEFT JOIN nh_clinical_spell AS spell '
```
Comment (Gimpneek) [inline nh_eobs_mental_health/models/nh_clinical_sql.py]: Joining with the spell and PME tables so I can then conditionally join with the refused table (using the spell.move_date to filter out transfers and the date_terminated of the last PME for obs_stop/restart)

File: nh_eobs_mental_health/models/nh_clinical_sql.py [hunk 5]
```diff
@@ -367,29 +367,44 @@ def get_refused_wardboard(self, interval):
...
+            'ON spell.activity_id = spell_activity.id '
+            'LEFT JOIN last_finished_pme AS pme ON pme.spell_id = spell.id '
             'LEFT JOIN refused_last_ews '
-            'ON refused_last_ews.spell_activity_id = spell.id '
+            'ON refused_last_ews.spell_activity_id = spell_activity.id '
+            'AND coalesce(refused_last_ews.date_terminated '
```
Comment (Gimpneek) [inline nh_eobs_mental_health/models/nh_clinical_sql.py]: the coalesce here says if there's no spell.move_date then return True as with the Demo data the move_date on the spell wasn't set, this shouldn't however be an issue on Live as the placement will set the move_date

File: nh_eobs_mental_health/models/nh_clinical_sql.py [hunk 6]
```diff
@@ -367,29 +367,44 @@ def get_refused_wardboard(self, interval):
...
         )
         return sql.format(activity_ids=activity_ids_sql)
 
     def get_collect_patients_sql(self, spell_ids):
         sql = self.collect_patients_skeleton.replace(
             'left join ews0 on ews0.spell_activity_id = activity.id',
             'left join ews0 on ews0.spell_activity_id = activity.id '
-            'left join refused_last_ews '
-            'on refused_last_ews.spell_activity_id = activity.id'
+            'LEFT JOIN nh_clinical_spell AS spell '
```
Comment (Gimpneek) [inline nh_eobs_mental_health/models/nh_clinical_sql.py]: I'd like to have been able to abstracted this out to be less DRY but there's a few differences (mostly the patient view uses the spell_activity_id (what was with them using that?))

File: nh_eobs_mental_health/tests/nh_eobs_api/test_get_activities_refused.py [hunk 7]
```diff
@@ -29,18 +35,23 @@ def patch_settings_activity_period(*args, **kwargs):
...
         super(TestGetActivitiesRefused, self).tearDown()
 
-    def get_refusal_in_effect(self, obs_data):
-        _logger.info('Completing first obs')
-        self.complete_obs(obs_data)
-        self.get_obs()
-        _logger.info('Completing refused obs')
-        self.complete_obs(clinical_risk_sample_data.REFUSED_DATA)
-        self.get_obs()
+    def get_refusal_in_effect(self, obs_list=None, patient_id=None):
```
Comment (Gimpneek) [inline nh_eobs_mental_health/tests/nh_eobs_api/test_get_activities_refused.py]: Made this method take a list of obs so can make things a little more uniform

**Comment by moderatemisbehaviour** [comment]
I'm happy with this :shipit:

**Comment by Nayira** [comment]
Tests for EOBS-595, including feature branch bugs EOBS-742, EOBS-744 (including fix for EOBS-730), EOBs-763 and EOBS-765 are a Pass :shipit:

**Pull request merged**